tests/bsim/bluetooth/mesh/src/test_beacon.c

```diff
@@ -1695,6 +1695,68 @@ static void test_rx_priv_multi_net_id(void)
 
 	PASS();
 }
+
+static void test_tx_priv_gatt_proxy(void)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &prb_comp);
+	provision(&rx_cfg);
+	bt_mesh_iv_update_test(true);
+
+	ASSERT_TRUE(bt_mesh.iv_index == 0);
+
+	/* Disable SNB. */
+	bt_mesh_beacon_set(false);
+	ASSERT_OK_MSG(bt_mesh_scan_disable(), "Failed to disable scanner");
+	ASSERT_OK_MSG(bt_mesh_gatt_proxy_set(BT_MESH_GATT_PROXY_DISABLED),
+		      "Failed to disable gatt proxy");
+	ASSERT_OK_MSG(bt_mesh_priv_gatt_proxy_set(BT_MESH_PRIV_GATT_PROXY_ENABLED),
+		      "Failed to set private gatt proxy");
+
+	/* Wait for proxy connection to complete. */
+	k_sleep(K_SECONDS(10));
+
+	/* Start IV update */
+	ASSERT_TRUE(bt_mesh_iv_update());
+
+	/* Check that IV index has updated */
+	ASSERT_TRUE(bt_mesh.iv_index == 1);
+	PASS();
+}
+
+static void test_rx_priv_gatt_proxy(void)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &prb_comp);
+	provision(&tx_cfg);
+	bt_mesh_iv_update_test(true);
```

Comment:
Why do you need IV index test mode if you initiate IV update procedure first time and only once?

Reply:
To be able to use `bt_mesh_iv_update()` I guess. I saw that it was used in other test and found that it made it quite easy to trigger the IV update. Does it matter? Since the test is not an IV update test I mean? I tested removing it only from the RX test, but then the IV index does not update.

Reply:
IV update is triggered without test mode. I guess this matters due to separate `if` for test mode. If you do not have plans to trigger IV update a couple of times for the short time you do not need test mode.

Reply:
Is there an easier way to trigger the IV update than this?

Reply:
```
bool bt_mesh_net_iv_update(uint32_t iv_index, bool iv_update);
```

Reply:
It does not work out of the box:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/35562897/2f32d0c7-2974-45aa-9f85-a9de89361f85)
Honestly I don't see the point of changing this from the original way if it entails more preconditioning and complexity just to avoid using the test API.

Reply:
Hm. Seems like a bug. If mesh didn't start IV update procedure before why minimum duration is checked. 
Ok, leave your previous variant, but this bug should be investigated separately.